sql/sys_vars.cc

```diff
@@ -2452,6 +2452,41 @@ Sys_gtid_ignore_duplicates(
        DEFAULT(FALSE), NO_MUTEX_GUARD,
        NOT_IN_BINLOG, ON_CHECK(check_gtid_ignore_duplicates),
        ON_UPDATE(fix_gtid_ignore_duplicates));
+
+static bool
+check_slave_max_statement_time(sys_var *self, THD *thd, set_var *var)
+{
+  return give_error_if_slave_running(0);
+}
+
+static bool
+update_slave_max_statement_time(sys_var *self, THD *thd, enum_var_type type)
+{
+  bool running;
+
+  mysql_mutex_unlock(&LOCK_global_system_variables);
+  running= give_error_if_slave_running(0);
+  mysql_mutex_lock(&LOCK_global_system_variables);
+  if (running) return running;
```

Comment:
@bnestere, please pay attention to [the coding style](https://docs.google.com/document/d/1CVz9BHss1kNIwu-Fz0ffKs67n9fy5zUVAo0Hgtw0x-8/edit#heading=h.z2s264aukuja). The if's branches must start on the if's following line.
Also it might be your editor does not always make the coding style's correct indentation. Will point to that specifically.